### ISSUE TITLE
installation error pytorch_lightning

### ISSUE THREAD

**Issue by arohami** [ChantalMP/RaDialog#8]
I am following the instructions mentioned [here](https://huggingface.co/ChantalPellegrini/RaDialog-interactive-radiology-report-generation) to try out the repo in colab. however, i get multiple erorrs in installation/setup phase:

1. after creating conda environment and activating it, installing requirements.txt throws error:
```
WARNING: Ignoring version 1.6.5 of pytorch_lightning since it has invalid metadata:
Requested pytorch_lightning==1.6.5 from https://files.pythonhosted.org/packages/34/de/3c98fb314e5c273a5c8bf0ff3b37e2a2625af7fb6540d9123cd5de975678/pytorch_lightning-1.6.5-py3-none-any.whl (from -r requirements.txt (line 7)) has invalid metadata: .* suffix can only be used with `==` or `!=` operators
    torch (>=1.8.*)
           ~~~~~~^
Please use pip<24.1 if you need to use this version
```
conda version is 23.11.0 if it helps. 
I also tested with pip==24.0 but there were other errors and warnings! So, I don't think the root issue is the pip. 

have any idea?

p.s: I have also followed instructions on github: RaDialog Environment > CheXbert Environment > Download pretrained models > activate RaDialog > Run Demo (skipping some steps) since i had many errors in other steps and because it seems that the demo code only needs radialog environment (it wasn't mentioned if the demo should be executed while radialog or chexbert is activated, so i went with radialog). this path also threw the same errors (one after each other after resolving each one) which I guess means that the problem is not the packages or pip version , etc.

Thanks very much

**Comment by ChantalMP** [comment]
Hi :)

Thanks for reaching out. I could reproduce the error you are describing - apparently pytorch-lightning==1.6.5 is not compatible with newer pip versions anymore.

I had to downgrade pip like you said with `pip install pip==24.0`
Additionally in the requirements.txt, you need to add `numpy==1.26.4`

Then during installation, the only warning I get is this, which also explains why a higher pip version does not work.
`DEPRECATION: pytorch-lightning 1.6.5 has a non-standard dependency specifier torch>=1.8.*. pip 24.1 will enforce this behaviour change. ...`

I can then successfully run the example code in the huggingface repo.

As reference, I am using the following versions:

conda: 22.9.0
Python: 3.10.15
pip: 24.0

I hope that helps and these fixes work for you as well! Let me know if you run into further errors, or need further information :)

Best, 
Chantal

**Comment by arohami** [comment]
@ChantalMP 
Thanks very much. I tried many ways and I kind of resolved the issue. However, I guess due to the limitations of google colab (free) i was not able to load the model. monitoring the gpu usage, i think it was overloaded and crashed. I switched to TPU but then some of the parts of the code didn't work with tpu. I reinstalled some of the packages and verified that the tpu was recognized (!tpu-info output).

in the end, I received another error in this part
```
import torch_xla.core.xla_model as xm
```
saying torch_xla doesn't have `core` attribute. and also this error:
```
ImportError _XLAC.cpython-310-x86_64-linux-gnu.so: undefined symbol: _ZN5torch3jit17parseSchemaOrNameERKSs
```
which made me realized it's not perhaps the best idea to try to make it work with tpu. 

anyways, thanks very much for the support. I decided not to continue working on these issues as I believe the main problem is memory limitation with T4 GPU. 

so, may I just ask about the requirements of the model? how much memory does it need?
